sdlib/d/gc/tcache.d

```diff
@@ -347,6 +377,29 @@ private:
 		return true;
 	}
 
+	void freeImpl(PageDescriptor pd, void* ptr) {
+		pd.arena.free(emap, pd, ptr);
+	}
+
+	void destroyImpl(PageDescriptor pd, void* ptr) {
+		auto e = pd.extent;
+		if (e is null) {
+			return;
+		}
+
+		if (pd.isSlab()) {
+			// Slab is not yet supported
+			return;
+		}
+
+		if (e.finalizer is null) {
+			return;
+		}
+
+		(cast(void function(void* body, size_t size)) e.finalizer)(
+			ptr, e.usedCapacity);
+	}
```

Comment:
I don't understand why this is destroyImpl rather than destroy itself. It has one callsite. Is it a generally useful abstraction?

Reply:
I still don't get why this isn't in the destroy method. What is the point of having two methods?